### PR TITLE
Laboratory work #1, Damir Khamidullin - 22FPL2

### PR DESCRIPTION
**Pull request by Damir-Khamidullin**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print('Hello, world!')
```
Comment (artyomtugaryov) [inline main.py:1]: please, remove the file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Nothing to review yet :(

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -37,6 +66,16 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
     :param actual: a list of actual values
     :return: the score
     """
+    if (type(predicted) != list or
```
[the remark this answers is not in the document]
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -19,6 +26,21 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if type(tokens) != list or tokens == []:
```
[the remark this answers is not in the document]
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -11,6 +11,13 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    tokens_list = []
+    if type(text) != str:
+        return None
+    for i in text:
```
[the remark this answers is not in the document]
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py:18]: done

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -11,6 +11,13 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    tokens_list = []
+    if type(text) != str:
```
[the remark this answers is not in the document]
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: main.py (hunk 1 above)
Comment (Damir-Khamidullin) [reply main.py:1]: deleted

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -11,6 +11,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    tokens_list = []
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:14]: create it after checks

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -19,6 +27,21 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if isinstance(tokens, list) is False or tokens == []:
+        return None
+    count = 0
+    for symbol in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:33]: too many loops

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -19,6 +27,21 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        if isinstance(symbol, str) is False:
+            count += 1
+    if count != 0:
+        return None
+    freq_dict = {}
+    for symbol in tokens:
+        if symbol in freq_dict:
+            freq_dict[symbol] += 1 / len(tokens)
+        else:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:42]: let's try without else

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -37,6 +67,16 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
     :return: the score
     """
+    if (isinstance(predicted, list) is False or
+            isinstance(actual, list) is False or
+            len(predicted) != len(actual)):
+        return None
+    mse_sum = 0
+    for i in range(len(predicted)):
+        mse_sum += (predicted[i] - actual[i])**2
+    mse = mse_sum / len(predicted)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:77]: do not create the variable

File: main.py (hunk 1 above)
Comment (artyomtugaryov) [reply main.py:1]: no

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 9 above)
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -11,6 +11,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    tokens_list = []
+    if isinstance(text, str) is False:
+        return None
+    text = text.lower()
+    for i in text:
```
[the remark this answers is not in the document]
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: main.py (hunk 1 above)
Comment (Damir-Khamidullin) [reply main.py]: is it deleted now?

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -11,6 +11,11 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if isinstance(text, str) is False:
+        return None
+    text = text.lower()
+    tokens_list = [i for i in text if i.isalpha()]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:17]: do not create the variable

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:35]: if `symbol` is in `freq_dict` - set 0 and always increment it

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -28,6 +44,13 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if isinstance(language, str) is False or isinstance(text, str) is False:
+        return None
+    language_profile = {"name": language,
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:49]: do not create the variable

File: lab_1_classify_by_unigrams/main.py [hunk 13]
```diff
@@ -73,6 +140,7 @@ def load_profile(path_to_file: str) -> dict | None:
     """
 
 
+
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:143]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 11 above)
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py (hunk 12 above)
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py (hunk 13 above)
Comment (Damir-Khamidullin) [reply lab_1_classify_by_unigrams/main.py]: changed

**Pull request closed without merging**